lib/modules/apostrophe-attachments/lib/api.js

```diff
@@ -605,4 +626,300 @@ module.exports = function(self, options) {
+
+    if (!doc.trash) {
+      commands.push([
+        {
+          _id: { $in: ids }
+        },
+        {
+          $addToSet: { docIds: doc._id }
+        }
+      ],
+      [
+        {
+          _id: { $in: ids }
+        },
+        {
+          $pull: { trashDocIds: doc._id }
+        }
+      ]);
+    } else {
+      commands.push([
+        {
+          _id: { $in: ids }
+        },
+        {
+          $addToSet: { trashDocIds: doc._id }
+        }
+      ],
+      [
+        {
+          _id: { $in: ids }
+        },
+        {
+          $pull: { docIds: doc._id }
+        }
+      ]);
+    }
+
+    commands.push([
+      {
+        $or: [
+          {
+            trashDocIds: { $in: [ doc._id ] }
+          },
+          {
+            docIds: { $in: [ doc._id ] }
+          }
+        ],
+        _id: { $nin: ids }
+      },
+      {
+        $pull: {
+          trashDocIds: doc._id,
+          docIds: doc._id
+        }
+      }
+    ], [
+      {
+        _id: { $in: ids }
+      },
+      {
+        $set: {
+          utilized: true
+        }
+      }
+    ]);
+
+    return async.series([
+      updateCounts,
+      hide,
+      show
+    ], callback);
+
+    function updateCounts(callback) {
+      return async.eachSeries(commands, function(command, callback) {
+        return self.db.update(command[0], command[1], callback);
+      }, callback);
+    }
+
+    function hide(callback) {
+      return self.db.find({
+        utilized: true,
+        'docIds.0': { $exists: 0 },
+        trash: { $ne: true }
+      }).toArray(function(err, attachments) {
+        if (err) {
+          return callback(err);
+        }
+        return async.eachSeries(attachments, hideOne, callback);
+      });
+    }
+
+    function show(callback) {
+      return self.db.find({
+        utilized: true,
+        'docIds.0': { $exists: 1 },
+        trash: { $ne: false }
+      }).toArray(function(err, attachments) {
+        if (err) {
+          return callback(err);
+        }
+        return async.eachSeries(attachments, showOne, callback);
+      });
+    }
+
+    function hideOne(attachment, callback) {
+      return permissionsOne(attachment, true, callback);
+    }
+
+    function showOne(attachment, callback) {
+      return permissionsOne(attachment, false, callback);
+    }
+
+    function permissionsOne(attachment, trash, callback) {
+
+      var method = trash ? self.uploadfs.disable : self.uploadfs.enable;
+      return async.series([
+        original,
+        scaled,
+        crops,
+        update
+      ], callback);
+
+      function original(callback) {
+        if ((!trash) && (attachment.trash === undefined)) {
+          // Trash status not set at all yet means
+          // it'll be a live file as of this point,
+          // skip extra API calls
+          return callback(null);
+        }
+        var path = self.url(attachment, { uploadfsPath: true, size: 'original' });
+        return method(path, function(err) {
```

Comment:
This might be redundant, you should be able to just treat `original` as a size, self.url knows what to do with that.